issue_4/contracts/GreenVault.sol

```diff
@@ -11,50 +22,82 @@ import "@yield-protocol/utils-v2/contracts/token/ERC20.sol";
 @notice A (single-token) vault that mints wrapped tokens
 in response to deposits
  */
-contract GreenVault is ERC20("GreenVaultToken", "OWL_GVT", 18) {
+contract GreenVault is
+    ERC20("GreenVaultToken", "OWL_GVT", 18),
+    Ownable,
+    FPMath
+{
     /**
     @notice the only token this vault supports
      */
-    IERC20 public theOnlySupportedToken;
+    IERC20Metadata public theOnlySupportedToken;
+
+    bool private is_entered;
```

Comment:
Do you really need reentrancy protection? I usually prefer to depend on a [checks-effects-interactions](https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html) pattern, and use mutex only if strictly necessary.

Reply:
Not really (since I do follow CEI), but I'm trying to use defensive programming
I'll take it out for now, but will research how expensive extra protection is